SalesforceCommerceCloud/pwa-kit
# @W-15439687 - Add default `stale_while_revalidate` cache control value

# Description

This PR adds a default `stale_while_revalidate` cache control header value. This will ensure cached responses will be returns to users even if they are expired, whist in the background the cache it update. 

I have chosen **1 year** as the length of time the stale response will be returned, but we can talk about changing that.

# Types of Changes


# Changes

- Add `STALE_WHILE_REVALIDATE` cache control header directive to templated files in the generator and home, plp, pdp in the template retail react app.

# How to Test-Drive This PR

- Open a terminal
- Run the following command **(ensure that you replace VALUE with a unique value, the result of Date.now() is good)**
```
curl -o /dev/null -s -w 'Total: %{time_total}s\n' -s "https://bchypak-production.mobify-storefront.com/global/en-GB/product/56736828M?mobify_cache_breaker=VALUE"
```
- Note the time it took to return that page, it should be in the 1-2s range.
- Make the exact same curl command and note the time should be closer to the 0.2-0.4s range (this is the cached value)
- You can make the request again and again to see the shortened response time.
- Wait 15 minutes (this is the cache expiration time).
- Make the same curl command again, and notice that it should take 0.2-0.4s, not 1-2s as you are being served the stale response. 
- If you wait  second and make the call again

## General
